pkg/api/v1/routes/handlers.go

```diff
@@ -161,34 +161,41 @@ func (r *Routes) serverEnroll(c *gin.Context) (int, *v1types.ServerResponse) {
 	}
 
 	// Creates a server record in FleetDB.
-	if err = r.fleetDBClient.AddServer(c.Request.Context(), serverID, params.Facility, params.IP, params.Username, params.Password); err != nil {
+	rollback, err := r.fleetDBClient.AddServer(c.Request.Context(), serverID, params.Facility, params.IP, params.Username, params.Password)
+	if err != nil {
+		rollbackErr := rollback()
 		if strings.Contains(err.Error(), badRequestErrMsg) {
 			return http.StatusBadRequest, &v1types.ServerResponse{
-				Message: err.Error(),
+				Message: err.Error() + fmt.Sprintf("server rollback err: %v", rollbackErr),
 			}
 		}
 		return http.StatusInternalServerError, &v1types.ServerResponse{
-			Message: err.Error(),
+			Message: err.Error() + fmt.Sprintf("server rollback err: %v", rollbackErr),
 		}
 	}
 
-	// Delete params related to add servers.
-	var inventoryArgs rctypes.InventoryTaskParameters
-	if err = json.Unmarshal(conditionCreate.Parameters, &inventoryArgs); err != nil {
-		return http.StatusBadRequest, &v1types.ServerResponse{
-			Message: "invalid inventory params: " + err.Error(),
-		}
+	inventoryArgs := &rctypes.InventoryTaskParameters{
+		AssetID:               serverID,
+		Method:                rctypes.OutofbandInventory,
+		CollectBiosCfg:        true,
+		CollectFirwmareStatus: true,
 	}
 	inventoryParams, err := json.Marshal(inventoryArgs)
```

Comment:
A minor thing, but I think it helps a bit: because we control `inventoryArgs` directly, we could (should?) panic if the serialization fails. Basically it should *never* fail, and if it did, there's not anything a caller of this API could do about it. If we instead `panic` here, we get an alert for ConditionAPI restarting, we get a stack trace, and likely we get more, better feedback than waiting for some random inquiry from support asking about why this customer can't enroll their server.

I wouldn't block merging this PR if you didn't want to do it now.

Reply:
Sent out #140